### PR TITLE
Clean up reminder tokens during time passes.

### PR DESCRIPTION
**Pull request by tomprince**
This is part of #3 with slightly cleaned up code.

Marking as draft, since I've not tested this version of the code yet.

### CODE REVIEW

File: global-script.lua [hunk 1]
```diff
@@ -1859,20 +1859,39 @@ function timePassesCo()
...
 end
+function getReminderColor(name)
+    return string.match(name, "^(%a*)'s %a* Reminder Tokens")
+end
 function handlePiece(object, depth)
-    if string.sub(object.getName(),1,4) == "City" then
+    local isoOffset = Vector(-11.2, 2, -1)
+    local defOffset = Vector(-9.5, 2, -1)
+    local name = object.getName()
+    local spos = getObjectFromGUID(elementScanZones["Red"]).getPosition()
```
Comment (iakona) [inline global-script.lua:1869]: what's the point of declaring the local variable here if you never use this value?

File: global-script.lua [hunk 2]
```diff
@@ -1859,20 +1859,39 @@ function timePassesCo()
...
         object = resetPiece(object, Vector(0,180,0), depth)
-    elseif string.sub(object.getName(),-7) == "Defence" then
+    elseif string.sub(name,-7) == "Defence" then
         if object.getLock() == false then object.destruct() end
         object = nil
+    elseif string.sub(name,-23) == "Isolate Reminder Tokens" then
+        if object.getLock() == false then
+             local color = getReminderColor(name)
+             spos = getObjectFromGUID(elementScanZones[color]).getPosition()
+             object.setPositionSmooth(Vector(spos.x,0,spos.z) + isoOffset)
```
Comment (iakona) [inline global-script.lua:1887]: this effectively will start stacking the tokens right?

File: global-script.lua (hunk 1 above)
Comment (tomprince) [reply global-script.lua:1869]: It is used in a later commit. I'll move it there.

File: global-script.lua (hunk 2 above)
Comment (tomprince) [reply global-script.lua:1887]: Yes.

**Comment by iakona** [comment]
closing this out since i've changed how these tokens work a bit, and wrote the code for cleaning them up too

**Comment by tomprince** [comment]
> closing this out since i've changed how these tokens work a bit, and wrote the code for cleaning them up too

I definitely want some kind of token to put on the spirit boards to track which growth and innate powers I've done, that gets cleaned up automatically. I was use the reminder tokens for this, particularly because they were handy. It looks like your patch to clean them up only does so on the sea tile.

**Comment by iakona** [comment]
hmm fair enough i'll reopen this PR and you can code up that part

**Comment by iakona** [comment]
ended up doing this myself https://github.com/spirit-island/spirit-island-mod/commit/6449171facaf5cd34c800176e639fad3267c6083

**Pull request closed without merging**